int8 state
int8 x_pos
int8 y_pos
